utils/handlers/handlers.go

```diff
@@ -113,3 +129,11 @@ func (fn AppHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		e.ServeHTTP(w, r)
 	}
 }
+
+// TransferRequestID transfers a request id from a context to a request header
+func TransferRequestID(ctx context.Context, r *http.Request) {
+	reqID, ok := ctx.Value(RequestID).(string)
+	if ok && len(reqID) != 0 {
+		r.Header.Set(requestIDHeaderKey, reqID)
+	}
+}
```

Comment:
This might make more sense if it was a stand-alone function within `utils.context` which merely returned the request id from the passed in context.  Something like this:
```go
func RequestIDFromContext(ctx context.Context) string {
    if reqID, ok := ctx.Value(RequestID).(string); ok {
        return reqID
    }
    return ""
}
```
Then in client.go instead of `handlers.TransferRequestID(ctx, req)` you could do `r.Header.Set(requestIDHeaderKey, context.RequestIDFromContext(ctx))`

I almost feel that this would be more consistent because we are already calling r.Header.Set a bunch in client.go.

Reply:
put fns in requestutils to avoid circular inputs